sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/AdtQuerySelectFrom.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Azure.DigitalTwins.Core.QueryBuilder
+{
+    /// <summary>
+    /// Query that contains a SELECT and FROM clause.
+    /// </summary>
+    internal class AdtQuerySelectFrom : AdtQuery
+    {
+        /// <summary>
+        /// Adds a WHERE and its conditional argument(s) clause to the query object. Meant to be used for simple
+        /// conditions involving operators or with basic ADT functions.
+        /// </summary>
+        /// <param name="condition"> A custom object that encodes the logical statement nested within the WHERE clause. </param>
+        /// <returns> The ADTQuery object itself. </returns>
+        internal AdtQuery Where(BaseCondition condition)
```

Comment:
Instead of AdtQuery return type, you may want to use `AdtQuerySelectFrom` so that you can keep appending where clauses.

Reply:
Look into how you are planning to use `BaseCoditionn`. ATM it seems like it's unnecessary, but I may be wrong. Also, the method accessibility seems wrong, methods should either be public/private/protected.

Reply:
Right now `ContainsCondition` and `ComparisonCondition` are both in `BaseCondition` to overload the `Where(Basecondition condition)` method in AdtQuerySelectFrom.cs -- however, turning that into two overloaded methods and changing them to be unrelated classes would work as well.